File: 9-1.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;

public class Solution {
    public static int translateDays(String x) {
        return Integer.parseInt(x.substring(0, 4)) * 12 * 28 + Integer.parseInt(x.substring(5, 7)) * 28 + Integer.parseInt(x.substring(8));
    }
    
    public int[] solution(String today, String[] terms, String[] privacies) {
        Map<String, Integer> T = new HashMap<>();
        for (String t : terms) {
            String[] parts = t.split(" ");
            String x = parts[0];
            String y = parts[1];
            T.put(x, Integer.parseInt(y) * 28);
        }
        List<Integer> answer = new ArrayList<>();
        int todayDays = translateDays(today);
        for (int i = 0; i < privacies.length; i++) {
            String[] parts = privacies[i].split(" ");
            String x = parts[0];
            String y = parts[1];
            int a = translateDays(x) + T.get(y);
            if (a <= todayDays) {
                answer.add(i + 1);
            }
        }
        int[] ans = new int[answer.size()];
        for (int i=0; i<ans.length; i++) ans[i]=answer.get(i);
        return ans;
    }
}

File: 9-1.py
def translate_days(x):
    return int(x[:4]) * 12 * 28 + int(x[5:7]) * 28 + int(x[8:])

def solution(today, terms, privacies):
    T = {}
    for t in terms:
        x, y = t.split()
        T[x] = int(y) * 28

    answer = []

    today = translate_days(today)

    for i in range(len(privacies)):
        x, y = privacies[i].split()

        a = translate_days(x) + T[y]
        if a <= today:
            answer.append(i + 1)

    return answer